app/controllers/HistoryController.scala

```diff
@@ -0,0 +1,23 @@
+package controllers
+
+import io.circe.syntax._
+import javax.inject.{Inject, Singleton}
+import models.services.HistoryControllerService
+import play.api.libs.circe.Circe
+import play.api.mvc._
+
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class HistoryController @Inject()(cc: ControllerComponents, hcs: HistoryControllerService)(implicit ex: ExecutionContext)
+  extends AbstractController(cc) with Circe {
+
+  def getHeaderR(id: String): Action[AnyContent] = Action.async {
+    hcs.getHeaderR(id).map ( a =>  Ok(a.asJson)).recover { case _ => BadRequest(s"Bad request") }
```

Comment:
Try to avoid using single letter variable name and also do not use such one-liners

Reply:
And also, `BadRequest(s"Bad request")` is a bit repetitive, you can just do `case _ => BadRequest `